### PR TITLE
fix: Handle duplicated change in manifest.appdescr_variant template for Adp Project 

### PR DESCRIPTION
**Pull request by GDamyanov**
fix for: #2080 

- handle duplicated change in manifest.appdescr_variant template for Adp Project

### CODE REVIEW

**Review by nikmace — APPROVED** [review]
Change looks straightforward
Did not test manually

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
TL;DR: I don't think your implementation does exactly what you intend, additionally, what you are trying to support here, should actually be fixed in the consumer and not here. Either fix the consumer, or do it generically.

In more detail:
* by default we are adding a change of the title which refers to the modified i18n model
* so we require it in the template an i18n model change as well
* if a consumer provides additional changes then, I agree, we shouldn't add duplicates, 
     * however you only check for the change type but not if it is actually the same model
     * or if someone adds a duplicate setTitle change
     * or if someone provides already duplicates with the content property

Please take the following content array for testing and modify your code so that you don't have duplicates
```
[
{ 
      "changeType": "appdescr_ui5_addNewModelEnhanceWith",
      "content": { 
              "modelId": "i18n|sap.suite.ui.generic.template.ObjectPage|ABC" },
              "texts": { "i18n": "i18n/ObjectPage/ABC/i18n.properties" }
},
{
      "changeType": "appdescr_app_setTitle",
      "content": {},
      "texts": {
        "i18n": "i18n/i18n.properties"
      }
}.
{ 
      "changeType": "appdescr_ui5_addNewModelEnhanceWith",
      "content": { 
              "modelId": "i18n|sap.suite.ui.generic.template.ObjectPage|ABC" },
              "texts": { "i18n": "i18n/ObjectPage/ABC/i18n.properties" }
}
]
